components/SaleorProviderWithChannels/SaleorProviderWithChannels.tsx

```diff
@@ -0,0 +1,23 @@
+import { SaleorProvider } from "@saleor/sdk";
+import React, { useEffect } from "react";
+
+import { saleorClient } from "@/lib/graphql";
```

Comment:
Bump @saleor/sdk to latest version (0.4.1), it exports a `useSaleorConfig` hook that has a `setChannel` method. Use that instead of importing the client itself

Reply:
Since this hook is not inside the Saleor provider, I cannot use the hook. 
Will back to it in separate PR.